sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -0,0 +1,250 @@
+| `EventProcessorHost()`| `EventHubConsumerClient(..., checkpoint_store)`| [receive events using checkpoint store](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhub/samples/async_samples/recv_with_checkpoint_store_async.py) |
+
+### Receiving events 
+
+| In v1                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.add_receiver()` and `Receiver.receive()`                       | `EventHubConsumerClient.receive()`                               | [receive events](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhub/samples/async_samples/recv_async.py) |
+
+### Sending events
+
+| In v1                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.add_sender()` and `Sender.send()`                          | `EventHubProducerClient.send_batch()`                               | [send events](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhub/samples/async_samples/send_async.py) |
+
+## Migration samples
+
+* [Receiving events](#migrating-code-from-eventhubclient-and-receiverasync-to-eventhubconsumerclient-for-receiving-events)
+* [Receiving events with checkpointing](#migrating-code-from-eventhubclient-and-senderasync-to-eventhubproducerclient-for-sending-events)
+* [Sending events](#migrating-code-from-eventprocessorhost-to-eventhubconsumerclient-for-receiving-events)
+
+### Migrating code from `EventHubClient` and `ReceiverAsync` to `EventHubConsumerClient` for receiving events
+
+In V1, `ReceiverAsync.receive()` returns a list of EventData.
+
+In V5, EventHubConsumerClient.receive() calls user callback on_event to process events.
+
+For example, this code which keeps receiving from a partition in V1:
+
+```python
+client = EventHubClientAsync.from_connection_string(connection_str)
+receiver = client.add_async_receiver(consumer_group="$default", partition="0", offset=Offset('@latest'))
+try:
+    await client.run_async()
+    logger = logging.getLogger("azure.eventhub")
+    while True:
+        received = await receiver.receive(timeout=5)
+        for event_data in received:
+            logger.info("Message received:{}".format(event_data.body_as_str()))
+finally:
+    await client.stop_async()
+```
+
+Becomes this in V5:
+
+```python
+logger = logging.getLogger("azure.eventhub")
+async def on_event(partition_context, event):
+    logger.info("Message received:{}".format(event.body_as_str()))
+
+client = EventHubConsumerClient.from_connection_string(
+    conn_str=CONNECTION_STR, consumer_group="$default", eventhub_name=EVENTHUB_NAME
```

Comment:
we should either put `eventhub` param into the `from_connection_string` in T1 or remove the `eventhub_name` param here to keep consistent.

both having eventhub or not:
```python
T1: client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
T2: client = EventHubConsumerClient.from_connection_string(
    conn_str=CONNECTION_STR, consumer_group="$default", eventhub_name=EVENTHUB_NAME
)
```

Reply:
Change to use eventhub=EVENTHUB_NAME for v1